pkg/systemd/hwinfo.jsx

```diff
@@ -192,173 +183,164 @@ class CPUSecurityMitigationsDialog extends React.Component {
+                rows={ sortedPci.map(dev => ({
+                    props: { key: dev.slot },
+                    columns: [dev.cls, dev.model, dev.vendor, dev.slot]
+                }))} />
+        );
+    }
 
-        return (
-            <Page>
-                <PageBreadcrumb stickyOnBreakpoint={{ default: "top" }}>
-                    <Breadcrumb>
-                        <BreadcrumbItem onClick={ () => cockpit.jump("/system", cockpit.transport.host)} className="pf-v5-c-breadcrumb__link">{ _("Overview") }</BreadcrumbItem>
-                        <BreadcrumbItem isActive>{ _("Hardware information") }</BreadcrumbItem>
-                    </Breadcrumb>
-                </PageBreadcrumb>
-                <PageSection>
-                    <Gallery hasGutter>
-                        <Card>
-                            <CardHeader>
-                                <CardTitle component="h2">{_("System information")}</CardTitle>
-                            </CardHeader>
-                            <CardBody>
-                                <SystemInfo info={this.props.info.system}
-                                            onSecurityClick={ this.state.mitigationsAvailable
-                                                ? () => Dialogs.show(<CPUSecurityMitigationsDialog />)
-                                                : undefined } />
-                            </CardBody>
-                        </Card>
-                        <Card id="pci-listing">
-                            <CardHeader>
-                                <CardTitle component="h2">{_("PCI")}</CardTitle>
-                            </CardHeader>
-                            <CardBody className="contains-list">
-                                { pci }
-                            </CardBody>
-                        </Card>
-                        <Card id="memory-listing">
-                            <CardHeader>
-                                <CardTitle component="h2">{_("Memory")}</CardTitle>
-                            </CardHeader>
-                            <CardBody className="contains-list">
-                                { memory }
-                            </CardBody>
-                        </Card>
-                    </Gallery>
-                </PageSection>
-            </Page>
+    if (info.memory.length > 0) {
+        memory = (
+            <ListingTable aria-label={ _("Memory") }
+                columns={ [_("ID"), _("Memory technology"), _("Type"), _("Size"), _("State"), _("Rank"), _("Speed")]}
+                rows={ info.memory.map(dimm => ({
+                    props: { key: dimm.locator },
+                    columns: [dimm.locator, dimm.technology, dimm.type, dimm.size, dimm.state, dimm.rank, dimm.speed]
+                })) } />
         );
+    } else if (!superuser.allowed) {
+        memory = (<EmptyState>
+            {_("Viewing memory information requires administrative access.")}
+        </EmptyState>);
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19378-20230922-161344-dd3ec708-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)